File: backend/app/schemas/artwork.py
from datetime import datetime
from typing import Literal, Optional
from pydantic import BaseModel, ConfigDict

ArtworkType = Literal["poster", "banner", "thumbnail"]


class ArtworkResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    episode_id: int
    artwork_type: str
    file_path: str
    width: Optional[int] = None
    height: Optional[int] = None
    file_size: Optional[int] = None
    mime_type: Optional[str] = None
    created_at: datetime
    updated_at: datetime

File: cms/src/types/artwork.ts
export type ArtworkType = 'poster' | 'banner' | 'thumbnail';

export interface Artwork {
  id: number;
  episode_id: number;
  artwork_type: ArtworkType;
  file_path: string;
  width: number | null;
  height: number | null;
  file_size: number | null;
  mime_type: string | null;
  created_at: string;
  updated_at: string;
}

export interface ArtworkSpec {
  type: ArtworkType;
  label: string;
  width: number;
  height: number;
  aspectRatio: string;
  maxSizeBytes: number;
}

export const ARTWORK_SPECS: Record<ArtworkType, ArtworkSpec> = {
  poster: {
    type: 'poster',
    label: 'Poster',
    width: 600,
    height: 900,
    aspectRatio: '2:3',
    maxSizeBytes: 200 * 1024,
  },
  banner: {
    type: 'banner',
    label: 'Banner',
    width: 1280,
    height: 720,
    aspectRatio: '16:9',
    maxSizeBytes: 200 * 1024,
  },
  thumbnail: {
    type: 'thumbnail',
    label: 'Thumbnail',
    width: 640,
    height: 360,
    aspectRatio: '16:9',
    maxSizeBytes: 200 * 1024,
  },
};
